ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp

```diff
@@ -284,8 +290,12 @@ OptimizedConvBlockConfig determine_per_core_conv_block_config(
             "Config Error: act_block_h_override must be a multiple of 32 (tile height).");
     }
     auto grid_size = parallel_config.grid.bounding_box().grid_size();
-    uint32_t act_block_h_ntiles = act_block_h_override > 0 ? act_block_h_override / 32
-                                                           : conv_op_parallel_config.per_core_out_matrix_height_ntiles;
+
+    uint32_t act_block_h_ntiles = conv_op_parallel_config.per_core_out_matrix_height_ntiles;
+    if (parallel_config.shard_scheme != TensorMemoryLayout::WIDTH_SHARDED && act_block_h_override > 0 ) {
+        act_block_h_ntiles = act_block_h_override / constants::TILE_HEIGHT;
```

Comment:
Why are we overriding the config values?

If we create the config values, shouldn't we fix that there?
If they are being passed in, shouldn't we fail the validation step and spell out in the documentation what is allowed and what isn't allowed?

Reply:
This is a part from width sharding hang issue. And this is being discussed there.
In this instance we need to ignore act_block_h_override is we are using width sharding and I agree,
we should document that it's going to be ignored with width sharding.

Even if we assert and say it's not allowed to set this property along with width sharding it's not going to be enough
since "auto sharding" feature could still select width sharding.

Reply:
Could you please provide a link to the discussion?

Should we log at least at the info level (not sure if warning is too much) that this is happening and the value should be changed?

Also, whatever code path is taken for the "auto sharding" feature, it should choose a proper config value if it doesn't already.

Reply:
Link to the relevant PR is in this PRs description.

Reply:
Didn't see it at first because the line numbers and code were different. Finally found it. https://github.com/tenstorrent/tt-metal/pull/13769/files#r1804309082
Thanks